crates/nri-resctrl-plugin/src/lib.rs

```diff
@@ -155,6 +232,111 @@ impl<P: FsProvider> ResctrlPlugin<P> {
             warn!("resctrl-plugin: failed to send event: {}", e);
         }
     }
+
+    // Create or fetch pod state and ensure group exists
+    fn handle_new_pod(&self, pod: &nri::api::PodSandbox) -> Option<String> {
+        let pod_uid = &pod.uid;
+        let mut st = self.state.lock().unwrap();
+        let ps = st.pods.entry(pod_uid.clone()).or_insert(PodState {
+            group_path: None,
+            total_containers: 0,
+            reconciled_containers: 0,
+        });
+        if ps.group_path.is_none() {
+            match self.resctrl.create_group(pod_uid) {
+                Ok(p) => {
+                    ps.group_path = Some(p.clone());
+                    Some(p)
+                }
+                Err(e) => {
+                    warn!("resctrl-plugin: failed to create group for pod {}: {}", pod_uid, e);
+                    None
+                }
+            }
+        } else {
+            ps.group_path.clone()
+        }
+    }
+
+    fn handle_new_container(&self, pod: &nri::api::PodSandbox, container: &nri::api::Container) {
+        let pod_uid = pod.uid.clone();
+        let group_path = match self.handle_new_pod(pod) {
```

Comment:
we assume the NRI infrastructure maintains the invariant that we see the pod notification before the container notification. We can check here and error! if the invariant isn't kept. Calling handle_new_pod is not warrented here.

But it makes sense to only attempt reconciliation if the pod state is successful -- if the resctrl group was never created, no use to try and reconcile